docs/src/references/tedge-configuration-management.md

```diff
@@ -0,0 +1,365 @@
+  and on change of the `plugins/tedge-configuration-plugin.toml` file.
+* Optional unix file ownership: `user`, `group` and octal `mode`.
+  These are only used when a configuration file pushed from a config_update command doesn't exist on the device.
+  When a configuration file is already present on the device, this plugin never changes file ownership,
+  ignoring these parameters.
+
+```sh
+cat /etc/tedge/plugins/tedge-configuration-plugin.toml
+```
+
+```toml title="file: /etc/tedge/plugins/tedge-configuration-plugin.toml"
+files = [
+  { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+  { path = '/etc/tedge/mosquitto-conf/c8y-bridge.conf' },
+  { path = '/etc/tedge/mosquitto-conf/tedge-mosquitto.conf' },
+  { path = '/etc/mosquitto/mosquitto.conf', type = 'mosquitto', user = 'mosquitto', group = 'mosquitto', mode = 0o644 }
+]
+```
+
+On start and when one of these files is updated, the configuration plugin sends
+a supported type declaration message with a retained flag
+to the config_snapshot and config_update command topics
+with the set of `type`s listed by the configuration
+(adding implicitly the `tedge-configuration-plugin` themselves).
+The message can be observed over the MQTT bus of the thin-edge device.
+
+Given that `root.topic` and `device.topic` are set to `te` and `device/main//` for the main device,
+the message to declare the supported configuration types is as follows.
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/config_snapshot' '{
+  "types": ["tedge-configuration-plugin", "tedge.toml", "/etc/tedge/mosquitto-conf/c8y-bridge.conf", "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf", "mosquitto"]
+}'
+```
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/config_update' '{
+  "types": ["tedge-configuration-plugin", "tedge.toml", "/etc/tedge/mosquitto-conf/c8y-bridge.conf", "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf", "mosquitto"]
+}'
+```
+
+:::note
+* The file `/etc/tedge/plugins/tedge-configuration-plugin.toml` itself doesn't need to be listed.
+  This is implied, so the list can *always* be configured from the configuration plugin.
+  The `type` for this self configuration file is `tedge-configuration-plugin`.
+* If the file `/etc/tedge/plugins/tedge-configuration-plugin.toml`
+  is not found, empty, ill-formed or not-readable
+  then only `tedge-configuration-plugin.toml` is declared as a supported configuration type.
+:::
+  
+The behavior of the `tedge-configuration-plugin` is also controlled
+by the configuration of thin-edge:
+
+* `tedge config get mqtt.bind.address`: the address of the local MQTT bus.
+* `tedge config get mqtt.bind.port`: the TCP port of the local MQTT bus.
+* `tedge config get root.topic`: the root of the [MQTT topic scheme](mqtt-api.md) to publish and subscribe.
+* `tedge config get device.topic`: the identifier of the [MQTT topic scheme](mqtt-api.md) to publish and subscribe.
+
+## Notifications (Do we need it? Read my opinion at the bottom of the file.)
```

Comment:
I agree with you, just wanted to hear other opinions. I would gladly delete this section.

Reply:
The command topics have a very different semantic meaning, compared to when you just want to notify some interested component that something has changed.

The command payload is specific for each command, so building a generic "file changed" notification is not possible as every interested component would have to know about which commands result in files being changed and the associated payload schema.

Instead it would be more useful to use one of the other telemetry topics to communicate the information, for example:

**Topic**

```text
te/device/main///status/file_changed
```

**Payload**
```sh
{"path":"/etc/tedge/mosquitto-conf/c8y-bridge.conf"}
```

Where the payload is associated with the `status/file_changed` schema.

Reply:
Yeah, agree that it'd be useful. But do we want to build this notification mechanism into the plugin, or as an external script, that can be used from other workflows as well?

Reply:
We can look at the notification functionality in a follow up step, as I'm pretty sure we'll find a more generic solution in the mqtt state machine work.